Brokerages/Zerodha/ZerodhaBrokerage.cs

```diff
@@ -0,0 +1,1386 @@
+        /// </summary>
+        public void UnlockStream()
+        {
+            Log.Trace("ZerodhaBrokerage.Messaging.UnlockStream(): Processing Backlog...");
+            while (_messageBuffer.Any())
+            {
+                MessageData e;
+                _messageBuffer.TryDequeue(out e);
+                OnMessageImpl(e);
+            }
+            Log.Trace("ZerodhaBrokerage.Messaging.UnlockStream(): Stream Unlocked.");
+            // Once dequeued in order; unlock stream.
+            _streamLocked = false;
+        }
+
+        #region IBrokerage
+
+        /// <summary>
+        /// Returns the brokerage account's base currency
+        /// </summary>
+        public override string AccountBaseCurrency => Currencies.INR;
+
+        /// <summary>
+        /// Checks if the websocket connection is connected or in the process of connecting
+        /// </summary>
+        public override bool IsConnected => WebSocket.IsOpen;
+
+        /// <summary>
+        /// Connects to Zerodha wss
+        /// </summary>
+        public override void Connect()
+        {
+            if (IsConnected)
+                return;
+
+            Log.Trace("ZerodhaBrokerage.Connect(): Connecting...");
+
+            var resetEvent = new ManualResetEvent(false);
+            EventHandler triggerEvent = (o, args) => resetEvent.Set();
+            WebSocket.Open += triggerEvent;
+            WebSocket.Connect();
+            if (!resetEvent.WaitOne(ConnectionTimeout))
+            {
+                throw new TimeoutException("Websockets connection timeout.");
+            }
+            WebSocket.Open -= triggerEvent;
+        }
+
+        /// <summary>
+        /// Closes the websockets connection
+        /// </summary>
+        public override void Disconnect()
+        {
+            //base.Disconnect();
+            if (WebSocket.IsOpen)
+            {
+                WebSocket.Close();
+            }
+        }
+
+
+        /// <summary>
+        /// Places a new order and assigns a new broker ID to the order
+        /// </summary>
+        /// <param name="order">The order to be placed</param>
+        /// <returns>True if the request for a new order has been placed, false otherwise</returns>
+        public override bool PlaceOrder(Order order)
+        {
+            LockStream();
+
+            uint orderQuantity = Convert.ToUInt32(Math.Abs(order.Quantity));
+            JObject orderResponse;
+            decimal? triggerPrice = null;
+
+            if (order.Status == OrderStatus.Invalid)
+            {
+                var errorMessage = $"Order failed, Order Id: {order.Id} timestamp: {order.Time} quantity: {order.Quantity} content: Invalid order status {order.Status}";
+                OnOrderEvent(new OrderEvent(order, DateTime.UtcNow.ConvertFromUtc(TimeZones.Kolkata), new OrderFee(new CashAmount(0, Currencies.INR)), "Zerodha Order Event") { Status = OrderStatus.Invalid });
```

Comment:
The order event time argument should always be UTC, 
same for all events in this method